internal/controller/thanosquery_controller.go

```diff
@@ -138,6 +141,7 @@ func (r *ThanosQueryReconciler) syncResources(ctx context.Context, query monitor
 			if err := ctrl.SetControllerReference(&query, obj, r.Scheme); err != nil {
 				r.logger.Error(err, "failed to set controller owner reference to resource")
 				errCount++
+				r.Recorder.Event(&query, corev1.EventTypeWarning, "SetOwnerReferenceFailed", fmt.Sprintf("Failed to set owner reference for resource %s: %v", obj.GetName(), err))
```

Comment:
i think this will have no impact on the end user really? and we continue on anyway